overlord/snapstate/handlers.go

```diff
@@ -3317,6 +3313,10 @@ func (m *SnapManager) doKillSnapApps(t *state.Task, _ *tomb.Tomb) (err error) {
 		return nil
 	}
 
+	st.Unlock()
+	defer st.Lock()
```

Comment:
Is this so that we match with `defer st.Unlock()` at the entry to the function?

Reply:
yes, I noticed in `stopSnapServices` and `undoStartSnapServices`, that `backend.StopServices` gets unlocked so I followed the pattern there. I can dig deeper into figuring out if the state unlock is ever needed.